provider/server/machine_delete_test.go

```diff
@@ -80,6 +195,18 @@ var _ = Describe("DeleteMachine", func() {
 		})
 		Expect(err).NotTo(HaveOccurred())
 
+		By("ensuring machine is in Terminating state after delete")
+		Eventually(func() iri.MachineState {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).Should(HaveLen(1))
+			return listResp.Machines[0].Status.State
+		}).WithTimeout(gracefulShutdownTimeout).WithPolling(1 * time.Second).Should(Equal(iri.MachineState_MACHINE_TERMINATING))
```

Comment:
i don't think, this is correct logic. You check state in loop and you end, if state is terminating, but state will be terminating immediately after send shutdown command. I recommend use ShutdownAt field in spec. ShutdownAt + gracefulShutdownTimeout is graceful shutdown period.